docs/source/guides/models.mdx

```diff
@@ -90,34 +130,51 @@ No worries, `NeuronModelForXXX` class will pad your inputs to an eligible shape.
 
 </Tip>
 
-Once your model is compiled, you can save it either on your local or in the [Hugging Face Model Hub](https://hf.co/models):
+## Export and inference of Generative NLP models
 
-```python
->>> from optimum.neuron import NeuronModelForSequenceClassification
+As explained in the previous paragraph, you will need only a few modifications to your Transformers code to export and run NLP models:
 
-# Load the model from the hub and export it to the Neuron optimized format
->>> input_shapes = {"batch_size": 1, "sequence_length": 64}
->>> model = NeuronModelForSequenceClassification.from_pretrained(
-...     "distilbert-base-uncased-finetuned-sst-2-english", export=True, **input_shapes
-... )
+```diff
+from transformers import AutoTokenizer
+-from transformers import AutoModelForCausalLM
++from optimum.neuron import NeuronModelForCausalLM
+
+# Instantiate and convert to Neuron a PyTorch checkpoint
+-model = AutoModelForCausalLM.from_pretrained("gpt2")
++model = NeuronModelForCausalLM.from_pretrained("gpt2", export=True)
+
+tokenizer = AutoTokenizer.from_pretrained("gpt2")
+tokenizer.pad_token_id = tokenizer.eos_token_id
+
+tokens = tokenizer("I really wish ", return_tensors="pt")
+with torch.inference_mode():
+    sample_output = model.generate(
+        **tokens,
+        do_sample=True,
+        min_length=128,
+        max_length=256,
+        temperature=0.7,
+    )
+    outputs = [tokenizer.decode(tok) for tok in sample_output]
+    print(outputs)
+```
 
-# Save the compiled model
->>> model.save_pretrained("a_local_path_for_compiled_neuron_model")
+### Configuring the export of a generative model
 
-# Push the onnx model to HF Hub
->>> model.push_to_hub(  # doctest: +SKIP
-...     "a_local_path_for_compiled_neuron_model", repository_id="my-neuron-repo", use_auth_token=True
-... )
-```
+There are three main parameters that can be passed to the `from_pretrained()` method to configure how a transformers checkpoint is exported to
+a neuron optimized model:
 
-And the next time when you want to run inference, just load your compiled model which will save you the compilation time:
+- `batch_size` is the number of input sequences that the model will accept,
+- `num_cores` is the number of neuron cores used when instantiating the model. Each neuron core has 16 Gb of memory, which means that
+bigger models need to be split on multiple cores,
+- `auto_cast_type` specifies the format to encode the weights. It can be one of `f32` (`float32`), `f16` (`float16`) or `bf16` (`bfloat16`).
```

Comment:
This also can be applied to other model. I just realized that I did not make a detailed doc for compiler's args for other models. I will make a PR for that as well.